### ISSUE TITLE
Implement data loader

### ISSUE THREAD

**Issue by HerrCooker** [es-ude/OnDeviceTraining#24]
- [x] load data sets from PyTorch
- [x] move `loadNPY()` to `Dataset.c`
- [x] add error, when drop last is set to false
- [x] check all files for `#define SOURCE_FILE ...`
- [x] implement shuffle
    - https://www.jstatsoft.org/article/download/v008i14/916 
---
- [x] move RNG out of dataloader
- [x] make one global `rng_t` instance
    - [x] can only be accessed/manipulated through `rngSetSeed()` and `rngGetSeed()`
    - This way `shuffleIndices()` can be used in other functions, e.g. init Layers with distributions
- [x] check if removing multiplication in `rngNext()` changes behaviour 

## Idea
- Use .npy files to export data sets form PyTorch
- parse .npy files in c
- https://jcastellssala.com/2014/02/01/npy-in-c/

---
## Reason for global datasets in UnitTest
- we want the signature of our getSample Function (which is externally passed and used for `getBatch()`) to only consist of id, and **NOT** any datasets, so the dataloader is completely seperate from any datasets
- i.e. this `getTrainSample(size_t id)`
- and not this `getSample(dataset_t *dataset, size_t id)`
- Furthermore we want to be able to pass datasets via USB, so the actual storage has to be completely independent from the actual dataloader

---
## Error in how we shuffle
- currently we shuffle batches
- we want to shuffle samples (before batching!), so inside the entire dataset
